src/components/share/share.js

```diff
@@ -23,23 +44,35 @@ class ShareControl extends BaseComponent {
   }
 
   /**
-   * before component mounted, set initial state
+   * toggle overlay internal component state
    *
    * @returns {void}
    * @memberof ShareControl
    */
-  componentWillMount() {
-    this.setState({overlay: false});
+  toggleOverlay(): void {
+    this.setState({overlay: !this.state.overlay});
+    this.props.toggleShareOverlay(this.state.overlay);
+    if (this.props.isPlaying || this.state.previousIsPlaying) {
+      this.setState({previousIsPlaying: true});
+    } else {
+      this.setState({previousIsPlaying: false});
+    }
+    if (this.state.overlay) {
+      this.player.pause();
+    } else if (this.state.previousIsPlaying) {
+      this.player.play();
```

Comment:
this logic should be applied in any overlay (advanced captions settings for example) no?

Reply:
No, it is defined to pause only on share overlay.
If this will become a generic requirement we can consider moving to overlay as a configurable option